### PR TITLE
✨ machineset: adjust preflight check to allow kubelet version skew of 3 for clusters running v1.28 and above

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

According to #8708 : Check and update the Kubernetes version skew policy in the MachineSet preflight checks.

This PR adjusts the machineset preflight checks to:

Allow minor version skew of 3 for kubelets, if the control plane is running kubernetes v1.28 and above.

See https://kubernetes.io/releases/version-skew-policy/#kubelet for more details.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/area machineset

### CODE REVIEW

File: internal/controllers/machineset/machineset_preflight.go [hunk 1]
```diff
@@ -173,7 +175,12 @@ func (r *Reconciler) kubernetesVersionPreflightCheck(cpSemver, msSemver semver.V
 	if msSemver.Minor > cpSemver.Minor {
 		return pointer.String(fmt.Sprintf("MachineSet version (%s) and ControlPlane version (%s) do not conform to the kubernetes version skew policy as MachineSet version is higher than ControlPlane version (%q preflight failed)", msSemver.String(), cpSemver.String(), clusterv1.MachineSetPreflightCheckKubernetesVersionSkew))
 	}
-	if msSemver.Minor < cpSemver.Minor-2 {
+	minorSkew := uint64(3)
```
Comment (sbueringer) [inline internal/controllers/machineset/machineset_preflight.go:178]: the godoc in l.173 is slightly off, let's replace that with something generic like "cannot be outside of the supported skew"

**Review by sbueringer — COMMENTED** [review]
Thx, just one nit

File: internal/controllers/machineset/machineset_preflight.go (hunk 1 above)
Comment (chrischdi) [reply internal/controllers/machineset/machineset_preflight.go:178]: Adjusted to inject the `minorSkew` variable instead of the hardcoded 2.

**Pull request merged**